### ISSUE TITLE
ExponentialSmoothing: downgrarde in model performance when migrating from V0.11.1 to V0.12.2

### ISSUE THREAD

**Issue by camilaagw** [statsmodels/statsmodels#7345]
#### Describe the bug

Currently, we are migrating our statsmodels version from 0.11.1 to 0.12.2. Nevertheless we observe that the upgrade causes a downgrade in the performance of one of our models.

#### Code Sample, a copy-pastable example if possible

Version 0.11.1:
```python
import pandas as pd
import statsmodels.tsa.holtwinters as hw
from sklearn.metrics import mean_absolute_percentage_error

series = pd.read_csv('https://raw.githubusercontent.com/unit8co/darts/master/examples/AirPassengers.csv')['#Passengers']
ts_pass_train, ts_pass_val = series[:97], series[97:]
hw_model = hw.ExponentialSmoothing(ts_pass_train.values,
                                           trend='additive',
                                           damped=False,
                                           seasonal='additive',
                                           seasonal_periods=12)

model = hw_model.fit() 
forecast = model.forecast(len(ts_pass_val))
mape = mean_absolute_percentage_error(ts_pass_val, forecast)
print(f'MAPE: {mape}')
forecast
```
![image](https://user-images.githubusercontent.com/22660466/108740133-a19bf080-7535-11eb-9482-ced24a8d9df7.png)

Version 0.12.2:
```python 
import pandas as pd
import statsmodels.tsa.holtwinters as hw
from sklearn.metrics import mean_absolute_percentage_error

series = pd.read_csv('https://raw.githubusercontent.com/unit8co/darts/master/examples/AirPassengers.csv')['#Passengers']
ts_pass_train, ts_pass_val = series[:97], series[97:]
hw_model = hw.ExponentialSmoothing(ts_pass_train.values,
                                           trend='additive',
                                           damped_trend=False,
                                           seasonal='additive',
                                           seasonal_periods=12)

model = hw_model.fit() 
forecast = model.forecast(len(ts_pass_val))
mape = mean_absolute_percentage_error(ts_pass_val, forecast)
print(f'MAPE: {mape}')
forecast
```
![image](https://user-images.githubusercontent.com/22660466/108740238-c1331900-7535-11eb-91a5-39ee6e9719af.png)

Note the only difference between both code snippets is the change of parameter  `damped` to `damped_trend`.

#### Expected Output

We expect to be able to achieve the same model performance in both versions.

#### Output of ``import statsmodels.api as sm; sm.show_versions()``

<details>

[paste the output of ``import statsmodels.api as sm; sm.show_versions()`` here below this line]

INSTALLED VERSIONS
------------------
Python: 3.8.5.final.0
OS: Darwin 19.6.0 Darwin Kernel Version 19.6.0: Sun Jul  5 00:43:10 PDT 2020; root:xnu-6153.141.1~9/RELEASE_X86_64 x86_64
byteorder: little
LC_ALL: None
LANG: None

statsmodels
===========

Installed: 0.12.2 (/Users/camilawilliamson/git/darts-env/lib/python3.8/site-packages/statsmodels)

Required Dependencies
=====================

cython: 0.29.17 (/Users/camilawilliamson/git/darts-env/lib/python3.8/site-packages/Cython)
numpy: 1.19.0 (/Users/camilawilliamson/git/darts-env/lib/python3.8/site-packages/numpy)
scipy: 1.5.0 (/Users/camilawilliamson/git/darts-env/lib/python3.8/site-packages/scipy)
pandas: 1.1.0 (/Users/camilawilliamson/git/darts-env/lib/python3.8/site-packages/pandas)
    dateutil: 2.8.1 (/Users/camilawilliamson/git/darts-env/lib/python3.8/site-packages/dateutil)
patsy: 0.5.1 (/Users/camilawilliamson/git/darts-env/lib/python3.8/site-packages/patsy)

Optional Dependencies
=====================

matplotlib: 3.2.2 (/Users/camilawilliamson/git/darts-env/lib/python3.8/site-packages/matplotlib)
    backend: module://ipykernel.pylab.backend_inline 
cvxopt: Not installed
joblib: 1.0.0 (/Users/camilawilliamson/git/darts-env/lib/python3.8/site-packages/joblib)

Developer Tools
================

IPython: 7.15.0 (/Users/camilawilliamson/git/darts-env/lib/python3.8/site-packages/IPython)
    jinja2: 2.11.2 (/Users/camilawilliamson/git/darts-env/lib/python3.8/site-packages/jinja2)
sphinx: 3.2.1 (/Users/camilawilliamson/git/darts-env/lib/python3.8/site-packages/sphinx)
    pygments: 2.7.4 (/Users/camilawilliamson/git/darts-env/lib/python3.8/site-packages/pygments)
pytest: Not installed
virtualenv: Not installed

</details>

**Comment by bashtage** [comment]
Check the SSE.  The new version achieves a much better SSE which is the in-sample fitting criteria.  That it happens to do worse at MAPE is just bad luck since MAPE is not the objective being optimized.

**Comment by bashtage** [comment]
0.11.1
```
res = hw_model.fit()
   ...: res.sse
C:\Anaconda\envs\sm11\lib\site-packages\statsmodels\tsa\holtwinters.py:743: ConvergenceWarning: Optimization failed to converge. Check mle_retvals.
  warn("Optimization failed to converge. Check mle_retvals.",
Out[3]: 10351.446452074531
```

0.12.2
```
In [13]: res.sse
Out[13]: 9784.991058807307
```

**Comment by camilaagw** [comment]
Thank you for following up on this @bashtage. The in-sample performance (sse) effectively improves with the version upgrade. However, when I check the mse for the validation set, I still get a downgrade in performance
```
from sklearn.metrics import mean_squared_error
mse = mean_squared_error(ts_pass_val, forecast)
```
0.11.1 MSE: 705.83
0.12.2  MSE: 1084.68
![image](https://user-images.githubusercontent.com/22660466/108844066-dfebeb00-75db-11eb-8ac2-889e1a464230.png)

I will try to experiment with other datasets, so see if I have the same case.

**Comment by bashtage** [comment]
The only thing that is guaranteed is the in-sample MSE fit, which is the objective.  The 0.12.2 version improves the in-sample MSE which is why there is no bug here.

**Comment by bashtage** [comment]
The fact that it does not converge should make one very skeptical of the 0.11.1 result. That it happened to forecast better in 0.11.1 is just luck.

**Comment by camilaagw** [comment]
You are right, I checked with other datasets and the new version seemed to be better for ExponentialSmoothing.

**Comment by bashtage** [comment]
Glad it works overall.
